command/acl_bootstrap.go

```diff
@@ -30,15 +30,19 @@ Bootstrap Options:
   -t
     Format and display the bootstrap response using a Go template.
 
+  -bootstrap-token
+    Provide an operator generated management token.
+
 `
 	return strings.TrimSpace(helpText)
 }
 
 func (c *ACLBootstrapCommand) AutocompleteFlags() complete.Flags {
 	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
 		complete.Flags{
-			"-json": complete.PredictNothing,
-			"-t":    complete.PredictAnything,
+			"-json":            complete.PredictNothing,
+			"-t":               complete.PredictAnything,
+			"-bootstrap-token": complete.PredictAnything,
```

Comment:
I feel like putting secrets on the command line is rarely the best option. Maybe @picatz can suggest a good standard approach for securely specifying secrets on a command line.

My thoughts are:
1. Keep this code. I don't think it's such a terrible idea we need to drop it.
2. Accept an `@filename` and/or `-` to read from stdin. Filename seems risky, but stdin might work really well for scripting.
3. `NOMAD_BOOTSTRAP_TOKEN="..."` env var to trigger the same behavior.

Also @picatz if I'm being overly cautious please say so! This feature is a *fantastic* quality of life improvement, and I really don't want to hold it up!

Reply:
I am happy to add extra options if needed.  I would need to find some examples of how best to do this and it could be implemented.
The standard method of a generated secret is received back in the terminal and so could have been captured in the terminal log as it is.  I would say that the risk is small if someone is running the client to set this. The operator running this command would need to secure the result for an autogenerated token. Our suggestion us always to use this token to create other tokens and then revoke it.
The ideal would be if they used terraform (after I update the provider)

Reply:
@picatz @schmichael are there any changes you want me to do here?

Reply:
Let's at least add `-bootstrap-token -` (note the trailing `-`) as a signal to read a token from stdin until newline.

That will allow both humans and scripts running the command to at least keep the token out of `ps aux` output.

Reply:
Thank you .

Reply:
> On that note, I imagine we should enforce some sort of lower-limit for this secret? For example: it needs to be at least 36 characters long.

We're validating that it's a UUID. (ref https://github.com/hashicorp/nomad/pull/12520/files#diff-28b31419c75cf45c1b9bf98fdc46721bb81a71c52e3004e0a01166f4230e561fR402-R409)

Reply:
> On that note, I imagine we should enforce some sort of lower-limit for this secret? For example: it needs to be at least 36 characters long.

We enforce the secret is a UUID, which while a bit strange, at least makes `@` detection easy (as its not a valid UUID character) and avoids other issues like complexity checks.

Reply:
Would it not be better to just remove the -bootstrap-token flag completely and then just accept 
- No args (Standard generated token)
- stdin read from there
- filename to read a file

Reply:
Yeah! That works but use `-` as the "filename" for stdin so the cases are:

```
nomad acl bootstrap   # same as today
nomad acl bootstrap - # read from stdin
nomad acl bootstrap root.token # read from file root.token
```

No big deal if you want to keep the flag though. Since folks should only be running this once-per-cluster I don't want to spend too much time optimizing the ergonomics. My main concern is security which we've already covered. :+1:

Reply:
It makes it easier to code if I remove the flag. for me at least.

I just need to fix the tests and then I can update the documents and we are good for this iteration.